File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <unistd.h>
#include <string.h>
#include <math.h>

float count_of_letters(string chrctrs);
float count_of_words(string chrctrs);
float count_of_sentances(string chrctrs);
float score(float lets, float wrds, float sents);

int main(void)
{
    string user_input = get_string("Text: \n");
    
    // count characters in user input
    float letters = count_of_letters(user_input);
    
    //count words in user input
    float words = count_of_words(user_input);
    
    //count sentances in user input
    float sentances = count_of_sentances(user_input);
    
    
    int answer = round(score(letters, words, sentances));
    
    if (answer>= 16)
    {
        printf("Grade 16+\n");
    }
    else if (answer < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", answer);
    }
}

float count_of_letters(string chrctrs)
{
    float letter_count = 0;
    
    for (int i=0, s = strlen(chrctrs); i< s; i++)
    {
        float temp = chrctrs[i];
        
        if (temp >= 65 && temp <= 90)
        {
            letter_count++;
        }
        else if (temp >= 97 && temp <= 122)
        {
            letter_count++;
        }
        
    }
    
    return letter_count;
}

float count_of_words(string chrctrs)
{
    float word_count = 1;
    
    for (int i=0, s = strlen(chrctrs); i< s; i++)
    {
        int temp = chrctrs[i];
        
        if (temp == 32)
        {
            word_count++;
        }
        
    }
    
    return word_count;
}

float count_of_sentances(string chrctrs)
{
    float sent_count = 0;
    
    for (int i=0, s = strlen(chrctrs); i< s; i++)
    {
        int temp = chrctrs[i];
        
        if (temp == 33 || temp == 46 || temp == 63)
        {
            sent_count++;
        }
        
    }
    
    return sent_count;
}

float score(float lets, float wrds, float sents)
{
    float index;
    
    float L = (lets/wrds)*100;
    float S = (sents/wrds)*100;
    
    index = 0.0588 * L - 0.296 * S - 15.8;
    
    return index;
}

File: readability.py
def get_letters(input):
    sent = 0
    wrds = 1
    let = 0
    i = 0

    while i < len(text):
        l = ord(text[i])

        # get words
        if l == 32:
            wrds += 1

        # get letters
        if l >= 65 and l <=90:
            let += 1
        elif l >= 97 and l <= 122:
            let += 1

        # get sentances
        if l == 33 or l == 46 or l == 63:
            sent += 1
        i += 1

    return(let, wrds, sent)


def reading_level(lets, words, sents):
    L = (lets/words)*100
    S = (sents/words) * 100

    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index >= 16:
        return("Grade 16+")
    elif index < 1:
        return("Before Grade 1")
    else:
        return(f"Grade {index}")


text = str(input("Text: "))

letter_count, word_count, sent_count = get_letters(text)

answer = reading_level(letter_count, word_count, sent_count)

print(answer)
